### ISSUE TITLE
Cloudy fog effect

### ISSUE THREAD

**Issue by KuriMC606** [devendrn/newb-x-mcbe#17]
New fog type with nlRenderVolFog function, uses ray matching on this fog to appear randomly and move on the map, similar to Rain Mist and this photo
![Volumetric-ground-fog](https://github.com/devendrn/newb-x-mcbe/assets/148877568/8d90ddf8-4e43-479f-ae71-4ae6412114ee)

**Comment by RukiMC123** [comment]
Is this possible?

**Comment by devendrn** [comment]
Would be quite hard to make something similar to that

**Comment by devendrn** [comment]
This is a good suggestion, but I'm not sure how the result will turn out to be. Two ways I think this could be implemented are:

Approach 1 (Realistic):
1. Use worldPos and cPos to somehow get tiledWorldPos. 
2. Create a 3D tiled noise pattern (add linear waves to get motion)
3. Add fog fade to 3d noise
4. Raymarch volumetrics
5. Mix with actual fog

Approach 2 (Just fake it):
1. Make some good looking noise pattern
2. Add it to fog

Approach 1 isn't really suited for this shader, but it's possible.

**Comment by devendrn** [comment]
>  Approach 1 is possible to do in vertex shader?

Yes, its possible. But it will look a little weird when you move because of interpolation. As example, look at the cloud/aurora reflection on water in this shader.

> how about for() function?

Yes you can use any control statement in vertex shader. See [GLES 2.0 Reference card](https://www.khronos.org/opengles/sdk/docs/reference_cards/OpenGL-ES-2_0-Reference-card.pdf) to understand GLSL more.

**Comment by devendrn** [comment]
Partially implemented.
